x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/url_state_actor.ts

```diff
@@ -35,7 +35,7 @@ export function createUrlInitializerActor({
     if (urlState.success) {
       // Always add default random samples data source
       if (!hasDefaultRandomSamplesDataSource(urlState.data.dataSources)) {
-        urlState.data.dataSources.unshift(defaultRandomSamplesDataSource);
+        urlState.data.dataSources.push(defaultRandomSamplesDataSource);
```

Comment:
When restoring data sources from the URL, push the default random samples source to the end to prioritize the URL ones